cedar-policy-core/src/parser/text_to_cst.rs

```diff
@@ -50,21 +50,23 @@ fn parse_collect_errors<'a, P, T>(
     let mut errs = Vec::new();
     let result = parse(parser, &mut errs, &Arc::from(text), text);
 
-    let mut errors: err::ParseErrors = errs
+    let errors = errs
         .into_iter()
         .map(err::ToCSTError::from_raw_err_recovery)
-        .collect();
+        .map(Into::into)
+        .collect::<Vec<err::ParseError>>();
```

Comment:
I think we can avoid this `.collect()` here and have it only in the `Err(e)` case.  The local variable `errors` here would just be an iterator.